source/Coop.Core/Client/Services/Towns/Handlers/ClientTownHandler.cs

```diff
@@ -78,7 +88,7 @@ private void HandleTownSecurity(MessagePayload<NetworkChangeTownSecurity> payloa
             messageBroker.Publish(this, message);
         }
 
-        private void HandleTownProsperity(MessagePayload<NetworkChangeTownProsperity> payload)
+        private void HandleProsperityTown(MessagePayload<NetworkChangeTownProsperity> payload)
```

Comment:
On the server there is : ```private void HandleTownProsperity(MessagePayload<TownProsperityChanged> obj)```
On the client : ```private void HandleTownProsperity(MessagePayload<NetworkChangeTownProsperity> payload)```

That's why I made the change, we should not subscribe with the same function name right ? Once I had a bug because of that, the message broker could not redirect to both function, only the last one register or the first one register was redirecting

Reply:
This should be fine, the messageBroker also stores the instance that owns the function. If this is truly and issue could you create a test that fails that case?

Reply:
Alright, there is no problem with the messageBroker, I reverted the changes. 
But there is clearly a bug with the town prosperity (same bug as Food Stock recently) 
```[HarmonyPatch(nameof(Town.Prosperity), MethodType.Setter)]
[HarmonyPrefix]
private static bool TownProsperityPrefix(ref Town __instance, ref float value)
```

This prefix is never trigger by the game, only by the command. So the game does not use the setter to change the town prosperity, and the Town Auditor show clearly this mismatch, so I'll create a new ticket for this bug